Automattic/wp-calypso
# Components: Refactor HEs to use react-query

#### Changes proposed in this Pull Request

Considering that the happiness engineers tray is a very straightforward example of server state on the client-side in Calypso, and the simplicity of the use case (read-only), this makes it a good example of how much boilerplate we can remove by using `react-query` instead of redux.

This PR refactors the happiness engineers tray Redux state to use `react-query` and ports the one single usage of this state tree. We're intentionally disabling `refetchOnWindowFocus` to preserve the previous behavior.

#### Testing instructions

* Go to `/devdocs/design/happiness-engineers-tray`
* Verify the HE avatars are properly loaded and displayed like before. Note that they're randomized on every re-render.
* Switch to another browser tab and come back to this one. 
* Verify that a new request was not issued to fetch HEs again, and the order is the same as before.
